### ISSUE TITLE
Build failure due to maven invoker plugin v3.2.2

### ISSUE THREAD

**Issue by rohanKanojia** [eclipse/jkube#1507]
### Description

Ever since we upgraded maven-invoker-plugin to v3.2.2 I'm seeing strange failures on CI. I'm also able to reproduce it on JDK8 on master branch :

```
openjdk version "1.8.0_332"
OpenJDK Runtime Environment (Temurin)(build 1.8.0_332-b09)
OpenJDK 64-Bit Server VM (Temurin)(build 25.332-b09, mixed mode)
```

When running `./mvnw test verify -B -Pjacoco`, I'm facing this error:
```sh
	at org.apache.maven.wrapper.MavenWrapperMain.main(MavenWrapperMain.java:61)
[INFO]   java.io.FileNotFoundException: /home/rokumar/work/repos/jkube/kubernetes-maven-plugin/it/target/it/xml-image-multilayer/target/docker/repository/xml-based-image-test/latest/build/Dockerfile (No such file or directory)
[INFO]           xml-image-multilayer/pom.xml ..................... FAILED (2.2 s)
[INFO] Building: xml-image/pom.xml
[INFO] [INFO] Scanning for projects...
[INFO] [WARNING] The POM for org.eclipse.jkube:kubernetes-maven-plugin:jar:1.8.0-SNAPSHOT is missing, no dependency information available
[INFO] [WARNING] Failed to retrieve plugin descriptor for org.eclipse.jkube:kubernetes-maven-plugin:1.8.0-SNAPSHOT: Plugin org.eclipse.jkube:kubernetes-maven-plugin:1.8.0-SNAPSHOT or one of its dependencies could not be resolved: Failure to find org.eclipse.jkube:kubernetes-maven-plugin:jar:1.8.0-SNAPSHOT in file:///home/rokumar/.m2/repository was cached in the local repository, resolution will not be reattempted until the update interval of local.central has elapsed or updates are forced
[INFO] [INFO] Downloading from local.central: file:///home/rokumar/.m2/repository/org/codehaus/mojo/maven-metadata.xml
[INFO] [INFO] Downloading from local.central: file:///home/rokumar/.m2/repository/org/apache/maven/plugins/maven-metadata.xml
[INFO] [INFO] Downloading from central: https://repo.maven.apache.org/maven2/org/codehaus/mojo/maven-metadata.xml
[INFO] [INFO] Downloading from central: https://repo.maven.apache.org/maven2/org/apache/maven/plugins/maven-metadata.xml
[INFO] [INFO] Downloaded from central: https://repo.maven.apache.org/maven2/org/codehaus/mojo/maven-metadata.xml (20 kB at 11 kB/s)
[INFO] [INFO] Downloaded from central: https://repo.maven.apache.org/maven2/org/apache/maven/plugins/maven-metadata.xml (14 kB at 7.5 kB/s)
[INFO] [INFO] ------------------------------------------------------------------------
[INFO] [INFO] BUILD FAILURE
[INFO] [INFO] ------------------------------------------------------------------------
[INFO] [INFO] Total time:  2.271 s
[INFO] [INFO] Finished at: 2022-05-19T19:50:39+05:30
[INFO] [INFO] ------------------------------------------------------------------------
[INFO] [ERROR] No plugin found for prefix 'k8s' in the current project and in the plugin groups [org.apache.maven.plugins, org.codehaus.mojo] available from the repositories [local (/home/rokumar/.m2/repository), local.central (file:///home/rokumar/.m2/repository), central (https://repo.maven.apache.org/maven2)] -> [Help 1]
[INFO] [ERROR] 
[INFO] [ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[INFO] [ERROR] Re-run Maven using the -X switch to enable full debug logging.
[INFO] [ERROR] 
[INFO] [ERROR] For more information about the errors and possible solutions, please read the following articles:
[INFO] [ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/NoPluginFoundForPrefixException
[INFO] run post-build script verify.groovy
[INFO] Running post-build script: /home/rokumar/work/repos/jkube/kubernetes-maven-plugin/it/target/it/xml-image/verify.groovy
[INFO] java.io.FileNotFoundException: /home/rokumar/work/repos/jkube/kubernetes-maven-plugin/it/target/it/xml-image/target/docker/repository/xml-based-image-test/latest/build/Dockerfile (No such file or directory)
	at java.io.FileInputStream.open0(Native Method)
	at java.io.FileInputStream.open(FileInputStream.java:195)
	at java.io.FileInputStream.<init>(FileInputStream.java:138)
	at org.eclipse.jkube.kit.common.ResourceVerify.readFile(ResourceVerify.java:88)
	at org.eclipse.jkube.kit.common.ResourceVerify.verifyContentEquals(ResourceVerify.java:70)
	at org.eclipse.jkube.maven.it.Verify.verifyContentEquals(Verify.java:46)
	at org.eclipse.jkube.maven.it.Verify$verifyContentEquals.call(Unknown Source)
	at org.codehaus.groovy.runtime.callsite.CallSiteArray.defaultCall(CallSiteArray.java:47)
	at org.codehaus.groovy.runtime.callsite.AbstractCallSite.call(AbstractCallSite.java:116)
	at org.codehaus.groovy.runtime.callsite.AbstractCallSite.call(AbstractCallSite.java:136)
	at Script1.run(Script1.groovy:19)
	at groovy.lang.GroovyShell.evaluate(GroovyShell.java:574)
	at groovy.lang.GroovyShell.evaluate(GroovyShell.java:612)
	at groovy.lang.GroovyShell.evaluate(GroovyShell.java:583)
	at org.apache.maven.shared.scriptinterpreter.GroovyScriptInterpreter.evaluateScript(GroovyScriptInterpreter.java:76)
	at org.apache.maven.shared.scriptinterpreter.ScriptRunner.executeRun(ScriptRunner.java:236)
	at org.apache.maven.shared.scriptinterpreter.ScriptRunner.run(ScriptRunner.java:161)
	at org.apache.maven.plugins.invoker.AbstractInvokerMojo.runPostBuildHook(AbstractInvokerMojo.java:2221)
	at org.apache.maven.plugins.invoker.AbstractInvokerMojo.runBuild(AbstractInvokerMojo.java:2196)
	at org.apache.maven.plugins.invoker.AbstractInvokerMojo.runBuild(AbstractInvokerMojo.java:1734)
	at org.apache.maven.plugins.invoker.AbstractInvokerMojo.runBuilds(AbstractInvokerMojo.java:1438)
	at org.apache.maven.plugins.invoker.AbstractInvokerMojo.execute(AbstractInvokerMojo.java:829)
	at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo(DefaultBuildPluginManager.java:137)
	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:210)
	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:156)
	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:148)
	at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject(LifecycleModuleBuilder.java:117)
	at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject(LifecycleModuleBuilder.java:81)
	at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build(SingleThreadedBuilder.java:56)
	at org.apache.maven.lifecycle.internal.LifecycleStarter.execute(LifecycleStarter.java:128)
	at org.apache.maven.DefaultMaven.doExecute(DefaultMaven.java:305)
	at org.apache.maven.DefaultMaven.doExecute(DefaultMaven.java:192)
	at org.apache.maven.DefaultMaven.execute(DefaultMaven.java:105)
	at org.apache.maven.cli.MavenCli.execute(MavenCli.java:957)
	at org.apache.maven.cli.MavenCli.doMain(MavenCli.java:289)
	at org.apache.maven.cli.MavenCli.main(MavenCli.java:193)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced(Launcher.java:282)
	at org.codehaus.plexus.classworlds.launcher.Launcher.launch(Launcher.java:225)
	at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode(Launcher.java:406)
	at org.codehaus.plexus.classworlds.launcher.Launcher.main(Launcher.java:347)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.apache.maven.wrapper.BootstrapMainStarter.start(BootstrapMainStarter.java:39)
	at org.apache.maven.wrapper.WrapperExecutor.execute(WrapperExecutor.java:122)
	at org.apache.maven.wrapper.MavenWrapperMain.main(MavenWrapperMain.java:61)
[INFO]   java.io.FileNotFoundException: /home/rokumar/work/repos/jkube/kubernetes-maven-plugin/it/target/it/xml-image/target/docker/repository/xml-based-image-test/latest/build/Dockerfile (No such file or directory)
[INFO]           xml-image/pom.xml ................................ FAILED (3.0 s)
[INFO] -------------------------------------------------
[INFO] Build Summary:
[INFO]   Passed: 0, Failed: 33, Errors: 0, Skipped: 0
[INFO] -------------------------------------------------
[ERROR] The following builds failed:
[ERROR] *  configmap/pom.xml
[ERROR] *  custom-environment/pom.xml
[ERROR] *  custom-raw-resources/pom.xml
[ERROR] *  deployment-strategy-type-919/pom.xml
[ERROR] *  docker-health-checks/pom.xml
[ERROR] *  env-metadata/pom.xml
[ERROR] *  fragments/pom.xml
[ERROR] *  helm-and-fragments/pom.xml
[ERROR] *  ingress-xml-config/pom.xml
[ERROR] *  java-options-env-merge/pom.xml
[ERROR] *  namespace/pom.xml
[ERROR] *  raw-resources/pom.xml
[ERROR] *  registry-286/pom.xml
[ERROR] *  remote-resources/pom.xml
[ERROR] *  revisionhistory/pom.xml
[ERROR] *  secret-config/pom.xml
[ERROR] *  sidecar/pom.xml
[ERROR] *  simple-maven-issue-mgmt/pom.xml
[ERROR] *  simple-maven-scm/pom.xml
[ERROR] *  simple-with-route-flag-false/pom.xml
[ERROR] *  simple-with-route-flag-true/pom.xml
[ERROR] *  simple/pom.xml
[ERROR] *  spring-boot/pom.xml
[ERROR] *  statefulset/pom.xml
[ERROR] *  vertx-default/pom.xml
[ERROR] *  vertx-health-check/pom.xml
[ERROR] *  vertx/pom.xml
[ERROR] *  volume-enricher-custom-storage-class/pom.xml
[ERROR] *  dependency-resources/pom.xml
[ERROR] *  probes-xml-config/pom.xml
[ERROR] *  volume-enricher-default-storage-class/pom.xml
[ERROR] *  xml-image-multilayer/pom.xml
[ERROR] *  xml-image/pom.xml
[INFO] -------------------------------------------------

```

I noticed this in https://github.com/eclipse/jkube/pull/1504, I suggest reverting back to 2.0.0

**Comment by manusa** [comment]
The version was bumped in #680.

The checks passed back then. The master branch CI pipelines pass intermittently.

Instead of reverting, it should be better to find out what's going on, especially since this was a major version bump. Probably there were some migration steps associated that we missed because the tests passed.

**Comment by manusa** [comment]
Maybe this is related to the JaCoCo agent, since failures happen only when using the jacoco profile (mvn clean install won't fail ---> CircleCI.test job fails, CircleCI.build job passes)

**Comment by manusa** [comment]
Closed by #1513
